pkg/limits/flags.go

```diff
@@ -0,0 +1,123 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license
+
+package limits
+
+import (
+	"flag"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/timescale/promscale/pkg/limits/mem"
+	"github.com/timescale/promscale/pkg/util"
+)
+
+var (
+	MemoryTargetMetric = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: util.PromNamespace,
+			Name:      "memory_target_bytes",
+			Help:      "The number of bytes of memory the system will target using",
+		})
+)
+
+type PercentageBytes struct {
+	percentage int    //integer [1-100], only set if specified as percentage
+	bytes      uint64 //always set
+}
+
+func (t *PercentageBytes) SetPercent(percent int) {
+	t.percentage = percent
+	t.bytes = 0
+}
+
+func (t *PercentageBytes) SetBytes(bytes uint64) {
+	t.percentage = 0
+	t.bytes = bytes
+}
+
+func (t *PercentageBytes) Get() (int, uint64) {
+	return t.percentage, t.bytes
+}
+
+func (t *PercentageBytes) Set(val string) error {
+	val = strings.TrimSpace(val)
+	percentage := false
+	if val[len(val)-1] == '%' {
```

Comment:
There are chances of corrupt inputs here. What happens if user inputs a decimal number? We wouldn't do that but shouldn't this be generic? (that's why the enum way)